genetic_algorithms/problems/base/state.py

```diff
@@ -17,4 +17,10 @@ def __str__(self) -> str:
     def __eq__(self, other):
         """
         Compares current states to another state.
+        """
+
+    @abstractmethod
+    def shuffle(self) -> 'State':
+        """
+        Shuffles state and returned an updated one
         """
```

Comment:
@mslazynski 
I am not sure if adding this method to state is good idea. 
The reason behind this is that I wanted to make a random restart in simulated annealing. 
I know that I could also do it by performing random moves from that state, but that that approach looked to me very similar to heating (because as far as I understood it, when we are increasing a temperature, we actually allowing our algorithm to perform random moves).
So i decided to add this method. 
What do you think about it?